278 student guide how to get started (#284)
ci-group/revolve2: docs/source/installation/index.rst

> Installation

Reviewer: Let's consistently write `Revolve2` capitalized everywhere and not mixed

Download your preferred version from `<https://github.com/ci-group/revolve2/releases>`_.
If you need to edit Revolve2 itself to add new features, it is recommended to instead create a fork and clone using git.

--------------------------------------------
Install packages
--------------------------------------------
Revolve2 contains multiple packages that provide specific functionality. These packages contain functionality for simulations, modular robot description, hardware control and optimization / EA.
If you need to edit components of revolve2, or you want to add new features, it is recommended that you use :ref:`installation/index:Editable Mode`.

> Manual installation with editable mode requires you to install the packages in order to avoid potential conflicts or missing dependencies.

Reviewer: ```suggestion
Manual installation with editable mode requires you to install the packages in order of dependency, so that all packages are installed in editable mode.
```

> For the correct order refer to the tables *requires* column.

Reviewer: ```suggestion
For the correct order refer to the tables *requires* column or look at the `dev_requirements.sh` script.
```

> Installing them without edible mode wont require the right order since it will automatically install dependencies.

Reviewer: ```suggestion
Installing them without edible mode does not require the right order since it will automatically install dependencies.
```

Each package can be installed using: ::

    pip install <package_name>

**For Students of the CI group:**
Packages with a **!** next to them are most likely required.
You can also use a shortcut to install all required packages: ::

>     sh student_install.sh

Reviewer: Shall we put this at the top of the section? so it's the first thing students read and do

     - :code:`modular_robot`
     - **!**
   * - modular_robot
     - This package provides revolve2 with all functionality around the robots and their modules.
     - :code:`simulation` & :code:`actor_controller`
     - **!**
   * - rpi_controller
     - This package allows to run a revolve2 ``ActorController`` on a physical robot, with the same behavior as in the simulations.
     - :code:`actor_controller`
     -
   * - rpi_controller_remote
     - This package allows to remotely connect to, and run a physical robot using SSH.
     - :code:`serialization`
     -
   * - serialization (Deprecated)
     - This package does what the name says. (Will be removed soon)
     -
     - **!**
   * - simulation
     - This package provides an abstraction layer for physics simulators. Other packages provide a simulator-specific implementation, such as for MuJoCo.
     - :code:`actor_controller`
     - **!**
   * - simulators/mujoco
     - This package provides simulation using the MuJoCo simulator.
     - :code:`simulation` & :code:`modular_robot`
     - **!**

-------------
Editable Mode
-------------

> When developing element in python packages it is crucial to test changes iteratively. To avoid constant :code:`pip uninstall` and :code:`pip install` spam, you can simply use pip`s built in "developer" mode.

Reviewer: ```suggestion
When developing element in python packages it is crucial to test changes iteratively. To avoid having to constantly :code:`pip uninstall` and :code:`pip install`, you can simply use pip`s built in "developer" mode.
```
